bottery/exceptions.py

```diff
@@ -0,0 +1,6 @@
+"""Exception classes for Bottery"""
+
+
+class ImproperlyConfigured(Exception):
+    """Bottery is somehow improperly configured"""
```

Comment:
I suggest creating a base exception class and make every bottery exception subclass that. For example, BotteryException.

Reply:
@nicoddemus why is that?

Reply:
Oh missed the previous discussions.

Having a base exception is usually beneficial to clients, because they can just catch `BotteryException` if they want to catch any error from the `bottery` framework.

```python

class BotteryException(Exception):
    """Base class Bottery exceptions"""

class ImproperlyConfigured(BotteryException):
    """Bottery is somehow improperly configured"""
```

And so on. This way you initially don't even need to actually specify a ton of exception subclasses, you can just raise `BotteryException`; if in the future you realize it is common for users to try to catch a specific error, you can create a subclass and existing clients will continue to work.

Reply:
Make total sense, thanks @nicoddemus, I'm going to open an issue for that.